api/src/main/java/org/openmrs/util/ThreadSafeCircularFifoQueue.java

```diff
@@ -756,15 +756,11 @@ public void remove() {
 				
 				if (prevIndex == NONE) {
 					throw new IllegalStateException();
-				} else if (prevIndex >= 0) {
-					if (elements[prevIndex] == prevItem) {
-						internalRemoveAtIndex(prevIndex);
-
-						if (prevIndex != read) {
-							nextIndex = Math.max(prevIndex, read);
-						}
-					}
-				}
+			  } else if ((prevIndex >= 0) && (elements[prevIndex] == prevItem)) {
+          internalRemoveAtIndex(prevIndex);
+        } else if (prevIndex != read) {
+          nextIndex = Math.max(prevIndex, read);
+        }
```

Comment:
And here.
![image](https://user-images.githubusercontent.com/9057882/91315589-250f9b80-e7b8-11ea-8523-581a99df3e1a.png)

Reply:
Finished